components/vni.js

```diff
@@ -0,0 +1,63 @@
+// Vni.js
+
+var _ = require('underscore');
+var Lm = require('./Lm');
+var state = require('./state');
+
+// Hash of vnis
+vnis = []; 
+
+// Define Vni class and its subclasses - InterNodeVni & IipVni
+function Vni( vnid ) {
+  this.vnid = vnid;
+  this.lm = Lm();
+  this.allInputStates = state.allInputStates;
+  this.allOutputStates = state.allOutputStates;
+  this.inputState = state.inputState;
+  this.outputState = state.outputState;
+  this.node = this.nodeInstance;
+}
+
+function NodePortVni( vnid ) {
+  Vni.call(this, vnid);
+  this.previousLms = [];
+}
+
+var IIP_VNID = '*';
```

Comment:
Followup: It works.  But you have to remember to explicitly create the intermediate hashes, unlike in Perl, which will create them automagically.